src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml.cpp

```diff
@@ -90,17 +90,29 @@ void EquationInputArea::AddNewEquation()
         return;
     }
 
-    m_lastLineColorIndex = (m_lastLineColorIndex + 1) % AvailableColors->Size;
-
     int colorIndex;
 
     if (m_accessibilitySettings->HighContrast)
     {
+        m_lastLineColorIndex = (m_lastLineColorIndex + 1) % AvailableColors->Size;
         colorIndex = m_lastLineColorIndex;
     }
     else
     {
-        colorIndex = colorAssignmentMapping[m_lastLineColorIndex];
+        std::array<bool, maxEquationSize> colorAssignmentUsed = { false, false, false, false, false, false, false,
```

Comment:
this is not necessary, C++ will zero-initialize all items if you write `std::array<bool, maxEquationSize> 
colorAssignmentUsed{}`

Reply:
Even if `maxEquationSize` is equals to the number of colors available, it can change in the future (limit the number of equations because of performances on some devices, etc...).

We should create another constant, only used for colors